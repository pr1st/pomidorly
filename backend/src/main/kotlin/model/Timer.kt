package model

import org.jetbrains.exposed.sql.Table

object Timers : Table() {
    val id = ActiveTasks.integer("id").primaryKey().autoIncrement()
    val pomidorDuration = integer("pomidor_duration")
    val shortBreakDuration = integer("short_break_duration")
    val longBreakDuration = integer("long_break_duration")
    val numberOfPomidorsBeforeLongBreak = integer("number_of_pomidors_before_long_break")
    val alarmWhenZero = bool("alarm_when_zero")
    val userId = (integer("user_id") references Users.id).uniqueIndex()
}

data class Timer(
    val id: Int,
    val pomidorDuration: Int,
    val shortBreakDuration: Int,
    val longBreakDuration: Int,
    val numberOfPomidorsBeforeLongBreak: Int,
    val alarmWhenZero: Boolean,
    val userId: Int
)

fun Timer.toDTO() =
    TimerDTO(pomidorDuration, shortBreakDuration, longBreakDuration, numberOfPomidorsBeforeLongBreak, alarmWhenZero)

data class TimerDTO(
    val pomidorDuration: Int,
    val shortBreakDuration: Int,
    val longBreakDuration: Int,
    val numberOfPomidorsBeforeLongBreak: Int,
    val alarmWhenZero: Boolean
)
