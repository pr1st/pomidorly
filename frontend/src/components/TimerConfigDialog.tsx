import React from 'react';

import "bootstrap/dist/css/bootstrap.css"
import {TimerConfigState} from "../types/timer";
import classNames from "classnames";


type Props = TimerConfigState & { setConfig: (timerConfig: TimerConfigState) => void }
type State = TimerConfigState

class TimerConfigDialog extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props)
        this.state = {
            ...props
        }
    }

    setParam = (paramName: string, value: number | boolean) => {
        if (paramName === "alarmWhenZero") {
            this.setState({
                ...this.state,
                alarmWhenZero: (value as boolean).valueOf()
            })
        } else {
            this.setState({
                    ...this.state,
                    [paramName]: (value as number).valueOf()
                }
            )
        }
    }

    save = () => {
        const setConfig = this.props.setConfig
        setConfig({
            ...this.state
        })
    }

    close = () => {
        this.setState({
            ...this.props
        })
    }

    render() {
        const {alarmWhenZero, longBreakDuration, numberOfPomidorsBeforeLongBreak, pomidorDuration, shortBreakDuration} = this.state

        return (
            <div id={"timerConfigModal"} className={classNames("modal", "fade")} tabIndex={-1}
            >
                <div className="modal-dialog">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title">Configuration properties</h5>
                            <button type="button" className="close" data-dismiss="modal" onClick={this.close} aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            <div>
                                <NumberParam name={"pomidorDuration"} label={"Pomidor duration"} value={pomidorDuration}
                                             onChange={this.setParam}/>
                            </div>
                            <div>
                                <NumberParam name={"shortBreakDuration"} label={"Short break duration"}
                                             value={shortBreakDuration} onChange={this.setParam}/>
                            </div>
                            <div>
                                <NumberParam name={"longBreakDuration"} label={"Long break duration"}
                                             value={longBreakDuration} onChange={this.setParam}/>
                            </div>
                            <div>
                                <NumberParam name={"numberOfPomidorsBeforeLongBreak"}
                                             label={"Number of pomidors before long break"}
                                             value={numberOfPomidorsBeforeLongBreak} onChange={this.setParam}/>
                            </div>
                            <div>
                                <BoolParam name={"alarmWhenZero"}
                                           label={"Alarm when zero"}
                                           value={alarmWhenZero}
                                           onChange={this.setParam}/>
                            </div>
                        </div>
                        <div className="modal-footer">
                            <button type="button" className="btn btn-primary" data-dismiss="modal" onClick={this.save}>Save changes</button>
                            <button id="closeDialog" type="button" className="btn btn-secondary" data-dismiss="modal" onClick={this.close}>Close</button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

function NumberParam(props: { name: string, label: string, value: number, onChange: (p: string, v: number) => void }) {
    const {name, value, onChange, label} = props
    return (
        <label>
            {label}:
            <input
                className={classNames("form-control")}
                value={value}
                type="number"
                onChange={
                    (e: { target: HTMLInputElement }) => {
                        const v = Number.parseInt(e.target.value)
                        if (v !== undefined) {
                            onChange(name, v)
                        }
                    }
                }
            />
        </label>
    )
}

function BoolParam(props: { name: string, label: string, value: boolean, onChange: (p: string, v: boolean) => void }) {
    const {name, value, onChange, label} = props
    return (
        <label>
            {label}
            <select
                value={value ? "true" : "false"}
                onChange={(e) => onChange(name, (e.target.value === "true"))}
                className={classNames("form-control")}>
                <option key={"true"} value={"true"}>{"true"}</option>
                <option key={"false"} value={"false"}>{"false"}</option>
            </select>
        </label>
    )
}

export default TimerConfigDialog;