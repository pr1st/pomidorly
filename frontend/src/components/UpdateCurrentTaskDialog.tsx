import React from 'react';
import "bootstrap/dist/css/bootstrap.css"
import {CurrentTaskState} from "../types/currentTasks";
import classNames from "classnames";

type Props = CurrentTaskState & { updateTask: (updateTask: CurrentTaskState) => void }
type State = CurrentTaskState

class UpdateCurrentTaskDialog extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            ...props
        }
    }

    setParam = (paramName: string, value: number | string) => {
        if (paramName === "numberOfPomidors") {
            this.setState({
                ...this.state,
                numberOfPomidors: (value as number).valueOf()
            })
        } else {
            this.setState({
                    ...this.state,
                    [paramName]: (value as string).valueOf()
                }
            )
        }
    };

    save = () => {
        this.props.updateTask({
            ...this.state
        })
    };

    close = () => {
        this.setState({
            ...this.props
        })
    };

    render() {
        const {numberOfPomidors, description, tag, id} = this.state;

        return (
            <div id={`updateCurrentTaskModal-${id}`} className={classNames("modal", "fade")} tabIndex={-1}
            >
                <div className="modal-dialog">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title">Update task</h5>
                            <button type="button" className="close" data-dismiss="modal" onClick={this.close}
                                    aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            <div>
                                <label>
                                    Id: {id}
                                </label>
                                <Param name={"tag"}
                                       label={"Tag"}
                                       value={tag}
                                       onChange={this.setParam}
                                />
                                <Param name={"description"}
                                       label={"Description"}
                                       value={description}
                                       onChange={this.setParam}
                                />
                                <Param name={"numberOfPomidors"}
                                       label={"Number of pomidors"}
                                       value={numberOfPomidors}
                                       onChange={this.setParam}
                                />
                            </div>
                        </div>
                        <div className="modal-footer">
                            <button type="button" className="btn btn-primary" data-dismiss="modal"
                                    onClick={this.save}>Save changes
                            </button>
                            <button type="button" className="btn btn-secondary" data-dismiss="modal"
                                    onClick={this.close}>Close
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

function Param(props: { name: string, label: string, value: number | string, onChange: (p: string, v: number | string) => void }) {
    const {name, value, onChange, label} = props
    return (
        <label className={classNames("Param")}>
            {label}:
            <input className={classNames("form-control")}
                   value={value}
                   type={name === "numberOfPomidors" ? "number" : "text"}
                   onChange={
                       (e: { target: HTMLInputElement }) => {
                           if (name === "numberOfPomidors") {
                               const v = Number.parseInt(e.target.value);
                               if (v !== undefined && v > 0) {
                                   onChange(name, v)
                               }
                           } else {
                               onChange(name, e.target.value)
                           }
                       }
                   }
            />
        </label>
    )
}

export default UpdateCurrentTaskDialog;