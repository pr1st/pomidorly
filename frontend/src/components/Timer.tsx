import React from 'react';
import {TimerConfigState, TimerState} from "../types/timer";
import classNames from "classnames";
import "bootstrap/dist/css/bootstrap.css"
import TimerConfigDialog from "./TimerConfigDialog";
import "bootstrap"

export type StateProps = TimerState

export type DispatchProps = {
    nextSecond: () => void,
    startTimer: () => void,
    pauseTimer: () => void,
    stopTimer: () => void,
    skipTimer: () => void,
    setConfig: (timerConfig: TimerConfigState) => void,
    doPomidor: () => void
}

let interval : NodeJS.Timeout;
let isRunning = false;

const Timer = ( props: StateProps & DispatchProps) => {
    const { nextSecond, pauseTimer, skipTimer, startTimer, stopTimer, setConfig, doPomidor} = props;
    const { numberOfPomidorsBeforeLongBreak,  alarmWhenZero} = props.config;
    const {isBreak, currentPomidor, time, isActive} = props.currentState;

    if (isActive && !isRunning) {
        isRunning = true;
        interval = setInterval(() => {
            clearInterval(interval);
            isRunning = false;
            if (time === 1) {
                if (alarmWhenZero) {
                    beepSound.play();
                }
                if (!isBreak) {
                    doPomidor()
                }
            }
            nextSecond()
        }, 1000)
     }
    const btnClass = classNames("btn", "btn-outline-primary", "btn-lg")
    const min = Math.floor(time/60)
    const sec = time % 60
    const actionState = isActive ? isBreak ? "Break" : "Action" : "Pause"

    return (
        <div className={classNames("Timer")}>
            <div className={"Options"}>
                <button
                    className={classNames(btnClass)}
                    data-toggle="modal"
                    data-target="#timerConfigModal"
                    data-backdrop="static"
                >
                    Config
                </button>
                <TimerConfigDialog {...props.config} setConfig={setConfig}/>
            </div>
            <div className={classNames("Time", actionState)}>{min}:{(sec < 10) ? "0"+sec : sec}</div>
            <h5>{actionState}</h5>
            <p>Pomidors: {currentPomidor}/{numberOfPomidorsBeforeLongBreak}</p>
            <div>
                <button
                    className={btnClass}
                    onClick={startTimer}
                >
                    Start
                </button>
                <button
                    className={btnClass}
                    onClick={pauseTimer}
                >
                    Pause
                </button>
                <button
                    className={btnClass}
                    onClick={stopTimer}
                >
                    Stop
                </button>
                <button
                    className={btnClass}
                    onClick={e => {
                        skipTimer()
                        if (!isBreak) {
                            doPomidor()
                        }
                    }}
                >
                    Skip
                </button>
            </div>
        </div>
    );
}

export default Timer;

// future technologies
const beepSound = new  Audio(
    "data:audio/wav;base64,//uQRAAAAWMSLwUIYAAsYkXgoQwAEaYLWfkWgAI0wWs/ItAAAG" +
    "DgYtAgAyN+QWaAAihwMWm4G8QQRDiMcCBcH3Cc+CDv/7xA4Tvh9Rz/y8QADBwMWgQAZG/" +
    "ILNAARQ4GLTcDeIIIhxGOBAuD7hOfBB3/94gcJ3w+o5/5eIAIAAAVwWgQAVQ2ORaIQwEMA" +
    "JiDg95G4nQL7mQVWI6GwRcfsZAcsKkJvxgxEjzFUgfHoSQ9Qq7KNwqHwuB13MA4a1q/Dm" +
    "BrHgPcmjiGoh//EwC5nGPEmS4RcfkVKOhJf+WOgoxJclFz3kgn//dBA+ya1GhurNn8zb//9" +
    "NNutNuhz31f////9vt///z+IdAEAAAK4LQIAKobHItEIYCGAExBwe8jcToF9zIKrEdDYIuP2MgO" +
    "WFSE34wYiR5iqQPj0JIeoVdlG4VD4XA67mAcNa1fhzA1jwHuTRxDUQ//iYBczjHiTJcIuPyKlHQkv" +
    "/LHQUYkuSi57yQT//uggfZNajQ3Vmz+Zt//+mm3Wm3Q576v////+32///5/EOgAAADVghQAAAAA//" +
    "uQZAUAB1WI0PZugAAAAAoQwAAAEk3nRd2qAAAAACiDgAAAAAAABCqEEQRLCgwpBGMlJkIz8jKhGvj4" +
    "k6jzRnqasNKIeoh5gI7BJaC1A1AoNBjJgbyApVS4IDlZgDU5WUAxEKDNmmALHzZp0Fkz1FMTmGFl1FM" +
    "EyodIavcCAUHDWrKAIA4aa2oCgILEBupZgHvAhEBcZ6joQBxS76AgccrFlczBvKLC0QI2cBoCFvfTDA" +
    "o7eoOQInqDPBtvrDEZBNYN5xwNwxQRfw8ZQ5wQVLvO8OYU+mHvFLlDh05Mdg7BT6YrRPpCBznMB2r//x" +
    "KJjyyOh+cImr2/4doscwD6neZjuZR4AgAABYAAAABy1xcdQtxYBYYZdifkUDgzzXaXn98Z0oi9ILU5m" +
    "BjFANmRwlVJ3/6jYDAmxaiDG3/6xjQQCCKkRb/6kg/wW+kSJ5//rLobkLSiKmqP/0ikJuDaSaSf/6JiLYLE" +
    "YnW/+kXg1WRVJL/9EmQ1YZIsv/6Qzwy5qk7/+tEU0nkls3/zIUMPKNX/6yZLf+kFgAfgGyLFAUwY//uQZAUABcd" +
    "5UiNPVXAAAApAAAAAE0VZQKw9ISAAACgAAAAAVQIygIElVrFkBS+Jhi+EAuu+lKAkYUEIsmEAEoMeDmCETMvfSHT" +
    "GkF5RWH7kz/ESHWPAq/kcCRhqBtMdokPdM7vil7RG98A2sc7zO6ZvTdM7pmOUAZTnJW+NXxqmd41dqJ6mLTXxrPpnV8ava" +
    "If5SvL7pndPvPpndJR9Kuu8fePvuiuhorgWjp7Mf/PRjxcFCPDkW31srioCExivv9lcwKEaHsf/7ow2Fl1T/9RkXgEhYElAoCLFt" +
    "MArxwivDJJ+bR1HTKJdlEoTELCIqgEwVGSQ+hIm0NbK8WXcTEI0UPoa2NbG4y2K00JEWbZavJXkYaqo9CRHS55FcZTjKEk3NKoCY" +
    "UnSQ0rWxrZbFKbKIhOKPZe1cJKzZSaQrIyULHDZmV5K4xySsDRKWOruanGtjLJXFEmwaIbDLX0hIPBUQPVFVkQkDoUNfSoDgQGKP" +
    "ekoxeGzA4DUvnn4bxzcZrtJyipKfPNy5w+9lnXwgqsiyHNeSVpemw4bWb9psYeq//uQZBoABQt4yMVxYAIAAAkQoAAAHvYpL5m6A" +
    "AgAACXDAAAAD59jblTirQe9upFsmZbpMudy7Lz1X1DYsxOOSWpfPqNX2WqktK0DMvuGwlbNj44TleLPQ+Gsfb+GOWOKJoIrWb3cI" +
    "MeeON6lz2umTqMXV8Mj30yWPpjoSa9ujK8SyeJP5y5mOW1D6hvLepeveEAEDo0mgCRClOEgANv3B9a6fikgUSu/DmAMATrGx7nng" +
    "5p5iimPNZsfQLYB2sDLIkzRKZOHGAaUyDcpFBSLG9MCQALgAIgQs2YunOszLSAyQYPVC2YdGGeHD2dTdJk1pAHGAWDjnkcLKFymS" +
    "3RQZTInzySoBwMG0QueC3gMsCEYxUqlrcxK6k1LQQcsmyYeQPdC2YfuGPASCBkcVMQQqpVJshui1tkXQJQV0OXGAZMXSOEEBRirX" +
    "bVRQW7ugq7IM7rPWSZyDlM3IuNEkxzCOJ0ny2ThNkyRai1b6ev//3dzNGzNb//4uAvHT5sURcZCFcuKLhOFs8mLAAEAt4UWAAIAB" +
    "AAAAAB4qbHo0tIjVkUU//uQZAwABfSFz3ZqQAAAAAngwAAAE1HjMp2qAAAAACZDgAAAD5UkTE1UgZEUExqYynN1qZvqIOREEFmBc" +
    "JQkwdxiFtw0qEOkGYfRDifBui9MQg4QAHAqWtAWHoCxu1Yf4VfWLPIM2mHDFsbQEVGwyqQoQcwnfHeIkNt9YnkiaS1oizycqJrx4" +
    "KOQjahZxWbcZgztj2c49nKmkId44S71j0c8eV9yDK6uPRzx5X18eDvjvQ6yKo9ZSS6l//8elePK/Lf//IInrOF/FvDoADYAGBMGb" +
    "7FtErm5MXMlmPAJQVgWta7Zx2go+8xJ0UiCb8LHHdftWyLJE0QIAIsI+UbXu67dZMjmgDGCGl1H+vpF4NSDckSIkk7Vd+sxEhBQM" +
    "RU8j/12UIRhzSaUdQ+rQU5kGeFxm+hb1oh6pWWmv3uvmReDl0UnvtapVaIzo1jZbf/pD6ElLqSX+rUmOQNpJFa/r+sa4e/" +
    "+NIEig0YfyWUho1VBBBA//uQZB4ABZx5zfMakeAAAAmwAAAAF5F3P0w9GtAAACfAAAAAwLhMDmAYWMgVEG1U0FIGCBgXBXAtfMH1" +
    "0000EEEEEECUBYln03TTTdNBDZopopYvrTTdNa325mImNg3TTPV9q3pmY0xoO6bv3r00y+IDGid/" +
    "/9Robv/Xx4v9ijkSRSNLQhAWumap82WRSBUqXStV/YcS+XVLnSS+WLDroqArFkMEsAS+eWmrUzrO0oEmE40RlMZ5+ODIkAyKAGUw" +
    "Z3mVKmcamcJnMW26MRPgUw6j+LkhyHGVGYjSUUKNpuJUQoOIAyDvEyG8S5yfK6dhZc0Tx1KI/gviKL6qvvFs1+bWtaz58uUNnryq" +
    "6kt5RzOCkPWlVqVX2a/EEBUdU1KrXLf40GoiiFXK///qpoiDXrOgqDR38JB0bw7SoL+ZB9o1RCkQjQ2CBYZKd/+VJxZRRZlqSkKi" +
    "ws0WFxUyCwsKiMy7hUVFhIaCrNQsKkTIsLivwKKigsj8XYlwt/WKi2N4d//uQRCSAAjURNIHpMZBGYiaQPSYyAAABLAAAAAAAACW" +
    "AAAAApUF/Mg+0aohSIRobBAsMlO//Kk4soosy1JSFRYWaLC4qZBYWFRGZdwqKiwkNBVmoWFSJkWFxX4FFRQWR+LsS4W/rFRb//////" +
    "//////////////////////////////////////////////////////////////////////////////////////////////////////" +
    "//////////////////////////////////////////////////////////////////////////////////////////////////////" +
    "//////////////////////////////////////////////////////////////////////////////////////////////////////" +
    "//////////////////////////////////////////////////////////////////////////////////////////////////////" +
    "///VEFHAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAU291bmRib3kuZGUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" +
    "AAAAAAAAAAAAAAAAAAAAAAAAAMjAwNGh0dHA6Ly93d3cuc291bmRib3kuZGUAAAAAAAAAACU=");
