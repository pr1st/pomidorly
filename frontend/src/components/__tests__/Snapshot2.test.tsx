import renderer from "react-test-renderer";
import Timer from "../Timer";
import TimerConfigDialog from "../TimerConfigDialog";
import UpdateCurrentTaskDialog from "../UpdateCurrentTaskDialog";
import React from "react";

describe("second snapshot test", () => {

    const dummy = () => {};


    it("Timer", () => {
        const tree = renderer
            .create(
                <Timer
                    doPomidor={dummy}
                    pauseTimer={dummy}
                    setConfig={dummy}
                    skipTimer={dummy}
                    startTimer={dummy}
                    stopTimer={dummy}
                    config={{
                        shortBreakDuration:1,
                        pomidorDuration:2,
                        alarmWhenZero:false,
                        longBreakDuration:3,
                        numberOfPomidorsBeforeLongBreak:5
                    }}
                    currentState={{
                        isActive:false,
                        currentPomidor:2,
                        startTime:123,
                        timeRemaining:3000,
                        isBreak:false
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("TimerConfigDialog", () => {
        const tree = renderer
            .create(
                <TimerConfigDialog
                    setConfig={dummy}
                    alarmWhenZero={false}
                    longBreakDuration={2}
                    numberOfPomidorsBeforeLongBreak={3}
                    pomidorDuration={5}
                    shortBreakDuration={123}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("UpdateCurrentTaskDialog", () => {
        const tree = renderer
            .create(
                <UpdateCurrentTaskDialog
                    updateTask={dummy}
                    id={1}
                    description={"ASD"}
                    tag={"ADSD"}
                    numberOfPomidors={2}
                    inQueue={3}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});