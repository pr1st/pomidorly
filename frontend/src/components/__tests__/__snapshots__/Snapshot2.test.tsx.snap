// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`second snapshot test Timer 1`] = `
<div
  className="Timer"
>
  <div
    className="Options"
  >
    <button
      className="btn btn-outline-primary btn-lg"
      data-backdrop="static"
      data-target="#timerConfigModal"
      data-toggle="modal"
    >
      Config
    </button>
    <div
      className="modal fade"
      id="timerConfigModal"
      tabIndex={-1}
    >
      <div
        className="modal-dialog"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
            >
              Configuration properties
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <label>
                Pomidor duration
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={2}
                />
              </label>
            </div>
            <div>
              <label>
                Short break duration
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={1}
                />
              </label>
            </div>
            <div>
              <label>
                Long break duration
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={3}
                />
              </label>
            </div>
            <div>
              <label>
                Number of pomidors before long break
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={5}
                />
              </label>
            </div>
            <div>
              <label>
                Alarm when zero
                <select
                  className="form-control"
                  onChange={[Function]}
                  value="false"
                >
                  <option
                    value="true"
                  >
                    true
                  </option>
                  <option
                    value="false"
                  >
                    false
                  </option>
                </select>
              </label>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Save changes
            </button>
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              id="closeDialog"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Time Pause"
  >
    0
    :
    03
  </div>
  <h5>
    Pause
  </h5>
  <p>
    Pomidors: 
    2
    /
    5
  </p>
  <div>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Start
    </button>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Pause
    </button>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Stop
    </button>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Skip
    </button>
  </div>
</div>
`;

exports[`second snapshot test TimerConfigDialog 1`] = `
<div
  className="modal fade"
  id="timerConfigModal"
  tabIndex={-1}
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Configuration properties
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div>
          <label>
            Pomidor duration
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={5}
            />
          </label>
        </div>
        <div>
          <label>
            Short break duration
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={123}
            />
          </label>
        </div>
        <div>
          <label>
            Long break duration
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={2}
            />
          </label>
        </div>
        <div>
          <label>
            Number of pomidors before long break
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={3}
            />
          </label>
        </div>
        <div>
          <label>
            Alarm when zero
            <select
              className="form-control"
              onChange={[Function]}
              value="false"
            >
              <option
                value="true"
              >
                true
              </option>
              <option
                value="false"
              >
                false
              </option>
            </select>
          </label>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Save changes
        </button>
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          id="closeDialog"
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`second snapshot test UpdateCurrentTaskDialog 1`] = `
<div
  className="modal fade"
  id="updateCurrentTaskModal-1"
  tabIndex={-1}
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Update task
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div>
          <label>
            Id: 
            1
          </label>
          <label
            className="Param"
          >
            Tag
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="text"
              value="ADSD"
            />
          </label>
          <label
            className="Param"
          >
            Description
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="text"
              value="ASD"
            />
          </label>
          <label
            className="Param"
          >
            Number of pomidors
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={2}
            />
          </label>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Save changes
        </button>
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
